comdb2/dbapi2.py

```diff
@@ -1105,7 +1176,13 @@ def _execute(self, operation, sql, parameters=None, *, column_types=None):
         try:
             # If variable interpolation fails, then translate the exception to
             # an InterfaceError to signal that it's a client-side problem.
-            sql = sql % {name: "@" + name for name in parameters}
+            # If binding by index then no need to modify sql
+            try:
+                by_name = hasattr(parameters, "items")
+            except Exception:
```

Comment:
What exception are we guarding against here? Is it a `NameError`?

Reply:
We don't know, it could be anything raised by `type(parameters).__getattr__`, with the exception of `AttributeError` (which `hasattr` would catch).

But I wasn't defensive like this in `_ccdb2.pyx`, so there's no point being defensive here. All it accomplishes is letting this pass 2 more layers down before failing. We might as well remove the `try` / `except` and let it fail here, instead.